src/app/view/perspectives/calendar/events/calendar-events.component.ts

```diff
@@ -141,9 +145,26 @@ export class CalendarEventsComponent implements OnInit, OnChanges {
   }
 
   public onNewEvent(time: number) {
-    const atLeastOneWritable = Object.values(this.permissions).some(p => p.writeWithView);
-    if (atLeastOneWritable) {
+    if (this.isAtLeastOneWritable()) {
       this.newEvent.emit(time);
     }
   }
+
+  private isAtLeastOneWritable(): boolean {
+    for (const entry of Object.entries(this.permissions)) {
+      if (entry[1].writeWithView && this.collectionHasConfig(entry[0])) {
```

Comment:
It is not very clear what `entry[0]` and `entry[1]` represent. Array destructuring might help here:
```typescript
for (const [collectionId, permissions] of Object.entries(this.permissions)) {
```

Reply:
Thanks!